trieste/space.py

```diff
@@ -999,6 +999,8 @@ class TaggedProductSearchSpace(CollectionSearchSpace):
         decision_space = Box([-1, -2], [2, 3])
         mixed_space = TaggedProductSearchSpace(spaces=[context_space, decision_space])
 
+    Note: the dtype of all the component search spaces must be the same.
```

Comment:
do we check for this and raise an Exception?

Reply:
Tensorflow will do that anytime we use `lower`, `sample` etc.